x/ibc-hooks/client/cli/query.go

```diff
@@ -42,26 +43,53 @@ func GetQueryCmd() *cobra.Command {
 	return cmd
 }
 
-// GetCmdPoolParams return pool params.
+// GetCmdWasmSender returns a generated local address for a wasm hooks sender.
 func GetCmdWasmSender() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "wasm-sender <channelID> <originalSender>",
 		Short: "Generate the local address for a wasm hooks sender",
 		Long: strings.TrimSpace(
 			fmt.Sprintf(`Generate the local address for a wasm hooks sender.
 Example:
-$ %s query ibc-hooks wasm-hooks-sender channel-42 juno12smx2wdlyttvyzvzg54y2vnqwq2qjatezqwqxu
+$ %s query ibchooks wasm-hooks-sender channel-42 juno12smx2wdlyttvyzvzg54y2vnqwq2qjatezqwqxu
 `,
 				version.AppName,
 			),
 		),
 		Args: cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
+			config := sdk.GetConfig()
 			channelID := args[0]
 			originalSender := args[1]
-			// ToDo: Make this flexible as an arg
-			prefix := sdk.GetConfig().GetBech32AccountAddrPrefix()
-			senderBech32, err := keeper.DeriveIntermediateSender(channelID, originalSender, prefix)
+
+			bech32PrefixArg, err := cmd.Flags().GetString(FlagBech32Prefix)
+			if err != nil {
+				return err
+			}
+
+			hashPrefixArg, err := cmd.Flags().GetString(FlagHashPrefix)
+			if err != nil {
+				return err
+			}
+
+			var hashPrefix string
+			if hashPrefixArg == "" {
+				hashPrefix = types.SenderPrefix
+			} else {
+				hashPrefix = hashPrefixArg
+			}
+
+			var bech32Prefix string
+			if bech32PrefixArg == "" {
+				bech32Prefix = config.GetBech32AccountAddrPrefix()
+			} else {
+				bech32Prefix, err = getBech32CustomPrefix(config, bech32PrefixArg)
```

Comment:
We can directly use the string supplied by the user as the prefix here. The idea is that you can use this to generate the derived addresses for other chains, so you can call `osmosisd q ibc-hooks wasm-hooks-sender --prefix=akash` and that would generate an akash-encoded address

Reply:
addressed